### ISSUE TITLE
Problem matching arguments of functions (parameter matching on function pointers)

### ISSUE THREAD

**Issue by albert-github** [doxygen/doxygen#10582]
Running the full worker 5.0.1 package, with 1.11.0 (1cbe096cf917e924082aebcefcef0927be26a3c1) I saw in "fieldlistview":
```
.../fieldlistview.cc:2473: warning: no matching class member found for
  void FieldListView::setSelectHandler(void(*nsh)(FieldListView *, int p))
Possible candidates:
  'void FieldListView::setSelectHandler(void(*nsh)(FieldListView *, int element))' at line 121 of file .../fieldlistview.h
  'void FieldListView::setSelectHandler(std::function< void(FieldListView *, int element) > cb)' at line 122 of file .../fieldlistview.h
```

Example: [example_4.tar.gz](https://github.com/doxygen/doxygen/files/14000989/example_4.tar.gz)

_Originally posted by @albert-github in https://github.com/doxygen/doxygen/issues/10559#issuecomment-1902593582_

**Comment by doxygen** [comment]
@albert-github I think I found a neat solution, which even works in case a function pointer parameter itself is again a function pointer.

**Comment by albert-github** [comment]
The warnings are gone, but in the output I see a bit of a strange place for the `nsh`:
![image](https://github.com/doxygen/doxygen/assets/5223533/baf0d91c-1034-486e-90ff-4af834052f68)

whilst in the source we see:
![image](https://github.com/doxygen/doxygen/assets/5223533/1051edc1-e7e3-4294-8889-b1f81c35a55d)

and the brief description:
![image](https://github.com/doxygen/doxygen/assets/5223533/b446f8ac-14bb-438d-b4d4-6e8104532d91)

**Comment by doxygen** [comment]
@albert-github The `nsh` placement I noticed as well, but it is unrelated to the matching issue and already exists for a long time. I need to look into this separately.

**Comment by albert-github** [comment]
@doxygen 
The placement of the `nsh` is indeed  a different topic, I think I saw it in some other cases as well but I can't quickly find the relevant issues.

**Comment by albert-github** [comment]
@doxygen

- I saw a fix for the "nsh" problem, though it is giving problems, at least, in the HTML output. For the test case 100 the difference is:
  **old**
  ```
          <td class="paramtype">void</td>          <td class="paramname"><span class="paramname"></span></td><td>)</td>
  ```
  **new**
  ```
          <td class="paramtype">void</td>          <td class="paramname"><span class="paramname"><em></em></span></em></span></td><td>)</td>
  ```

- Another thing I see is the usage of `<em>`, why?

**Comment by albert-github** [comment]
It looks like there s a problem with `(void)` in the as when removing the word `void` (in `100_a.c` and `more_100_c.c`) it looks OK, though when changing `void` into `int` on both places it gives a problem again.

**Comment by albert-github** [comment]
It looks like there is some interaction between the `endParameterName`  and the `startParameterExtra`, this behavior can be nicely seen in the latex output:
```
(\begin{DoxyParamCaption}\item[{void}]
%%start
{}%%end

%%start extra
{}%%end

%%start extra
{}%%end extra
\end{DoxyParamCaption})
```
when adjusting the latexgen with some comment:
```
void LatexGenerator::startParameterName(bool /*oneArgOnly*/)
{
  m_t << "\n%%start\n{";
}

void LatexGenerator::endParameterName()
{
  m_t << "}%%end\n";
}

void LatexGenerator::startParameterExtra()
{
  m_t << "\n%%start extra\n{";
}

void LatexGenerator::endParameterExtra(bool last,bool /*emptyList*/,bool closeBracket)
{
  m_t << "}%%end extra\n";
  if (last)
  {
    m_t << "\\end{DoxyParamCaption}";
    if (closeBracket) m_t << ")";
  }
}
```

Note it looks like the possible parameter of `startParameterName` is nowhere used.

**Comment by albert-github** [comment]
Looks OK with commit [9a9fff2](https://github.com/doxygen/doxygen/commit/9a9fff28c9ecf451f8538eb81594d3dabd1d9ee0)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
